studies/forms.py

```diff
@@ -223,7 +309,7 @@ class StudyEditForm(StudyForm):
     """Form for editing study"""
 
     def __init__(self, *args, **kwargs):
-        user = kwargs.get("user")
+        user = kwargs.pop("user")
```

Comment:
Why are we `pop`ing here? I've seen this in a couple of places. Is the mutation intentional? If it's not, better to use `get` or `brackets`

Reply:
Yes, intentional because we use `user` here to determine the lab queryset but don't want to include it in the call to `super().init`. This is also done e.g. in the new `TOTPCheckForm` so I thought it was ok?

Reply:
Uy, no 🙈  my bad.

TOTPCheckForm's initializer should be like so:
```
    def __init__(self, request, *args, **kwargs):
        self.request = request
        self.otp = getattr(request.user, "otp")
        super().__init__(*args, **kwargs)
```

Argument parsing shouldn't be done imperatively - let the signature itself decide what's expected at call time.